hr_holidays_leave_request_wizard/models/hr_leave.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2018-2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from pytz import timezone, UTC
+from datetime import time
+
+from odoo import api, fields, models
+
+
+class HrLeave(models.Model):
+    _inherit = 'hr.leave'
+
+    @api.model
+    def _default_get_request_parameters(self, values):
+        HrEmployee = self.env['hr.employee'].sudo()
+        tz = timezone(self.env.context.get('tz') or self.env.user.tz or 'UTC')
+
+        new_values = dict(values)
+        now = fields.Datetime.now()
+        date_from = values.get('date_from', now)
+        date_to = values.get('date_to', now)
+
+        if not date_from.tzinfo:
+            date_from = date_from.replace(tzinfo=UTC)
+        if not date_to.tzinfo:
+            date_to = date_to.replace(tzinfo=UTC)
+
+        local_date_from = date_from.astimezone(tz)
+        local_date_to = date_to.astimezone(tz)
+
+        default_time_from = time(7, 0, 0, 0)
+        default_time_to = time(19, 0, 0, 0)
+
+        if date_from == date_to:
+            # NOTE: Simulate 7am/7pm default from calendar view
+            local_date_from = local_date_from.replace(
+                hour=default_time_from.hour,
+                minute=default_time_from.minute,
+                second=default_time_from.second,
+                microsecond=default_time_from.microsecond,
+            )
+            date_from = local_date_from.astimezone(UTC)
+            local_date_to = local_date_to.replace(
+                hour=default_time_to.hour,
+                minute=default_time_to.minute,
+                second=default_time_to.second,
+                microsecond=default_time_to.microsecond,
+            )
+            date_to = local_date_to.astimezone(UTC)
+
+        if values.get('holiday_type', 'employee') == 'employee' \
+                and local_date_from.time() == default_time_from \
+                and local_date_to.time() == default_time_to:
+            employee_id = values.get(
+                'employee_id',
+                self.env.context.get('default_employee_id')
+            )
+            if employee_id:
+                employee = HrEmployee.browse(employee_id)
+            else:
+                user_id = values.get('user_id', self.env.user.id)
+                employee = HrEmployee.search([
```

Comment:
or this `employee = self.env.user.employee_ids[:1].id` ?

Reply:
user_id is dynamic here